### ISSUE TITLE
MIDI device selection issues in 0.54-0test1 (64bit windows, automatic installer)

### ISSUE THREAD

**Issue by jlsiegel07** [pure-data/pure-data#2005]
After running the new test version, I was unable to select my intended midi devices in the dialog options by selecting their names, would receive a message stating Port Midi Host Error.

I could occasionally get the intended behavior (receiving input from a nektar 66 keyboard on my desk attached by USB, sending output to a Focusrite audio/midi device also connected by USB), but only by selecting options in the dropdown list that are labelled as other sources (example: Selecting the Focusrite USB device as input would actually result in using the Nektar keyboard's midi as input, and selecting Microsoft MIDI Mapper for output would send to the Focusrite MIDI out).

In short, the options in the list do not correspond to what device PD actually attempts to use.

This is on the following windows version:
Edition	Windows 10 Pro
Version	22H2
Installed on	‎10/‎10/‎2020
OS build	19045.3086
Experience	Windows Feature Experience Pack 1000.19041.1000.0

**Comment by umlaeute** [comment]
confirmed in Wine

**Comment by Spacechild1** [comment]
Can confirm on Windows 10. Looks like the device index in the MIDI settings is off by 1. Also, it is strange that the device list contains both "none" (first entry) and "(kein Gerät)" (last entry):

![grafik](https://github.com/pure-data/pure-data/assets/16126632/f988081b-da37-4e33-a26b-71fa998a39a2)

**Comment by Spacechild1** [comment]
I get the same behavior on macOS. I guess this problem has been introduced with the audio/MIDI dialog revamp.

**Comment by umlaeute** [comment]
yikes. the numbering used to be super-confusing and I tried to fix it for good...

**Comment by Lucarda** [comment]
I'm lost here

https://github.com/pure-data/pure-data/blob/6451fe0f1cb7737623d1e3536d424699d6271364/src/s_midi.c#L718-LL722

the message is 5 args but 9 are given.

@umlaeute ?

**Comment by umlaeute** [comment]
@Lucarda https://github.com/pure-data/pure-data/blob/6451fe0f1cb7737623d1e3536d424699d6271364/src/m_pd.h#L784-L785

**Comment by Lucarda** [comment]
Thanks :)

if i use `+ 1` instead of "decrement values" i get the correct values when reading from the Registry but if I change the output setting and save I write "offsetted" to the registry (two devices away)

```
proc ::dialog_midi::set_configuration {API inputdevices indevs outputdevices  outdevs} {
    set ::pd_whichmidiapi $API

    set ::midi_indevlist ${inputdevices}
    # decrement indices by 1 (and filter out invalid(=negative) values)
    set ::midi_indevices [lmap _ $indevs {set _ [expr int($_) + 1]; if {$_ < 0} continue; set _}]

    set ::midi_outdevlist ${outputdevices}
    set ::midi_outdevices [lmap _ $outdevs {set _ [expr int($_) + 1]; if {$_ < 0} continue; set _}]
}
```
